pkg/middleware/logging.go

```diff
@@ -0,0 +1,37 @@
+// Copyright KubeArchive Authors
+// SPDX-License-Identifier: Apache-2.0
+package middleware
+
+import (
+	"log/slog"
+	"time"
+
+	"github.com/gin-gonic/gin"
+)
+
+func Logger(c *gin.Context) {
+	start := time.Now()
+	c.Next()
+	end := time.Now()
+
+	latency := end.Sub(start)
+	if latency > time.Minute { // Avoid very large latency number string
+		latency = latency.Truncate(time.Second)
+	}
```

Comment:
You should probably be using `Round` instead of `Truncate` here.

Why are you only truncating the latency if it is more than a minute? Comparing the results is going to look weird if some of the values have nano-second accuracy and some do not. As I understand it, the default display of duration (with a "latency" label) in Go looks like this:

`latency: 1m26.550805438s`

With your conditional Truncation (or Round), you will end up comparing strings like this:

```
latency: 1m10s
latency: 26.550805438s
```

If you are worried about the final string being too long, why not shorten everything to the same precision? If you did a `latency = latency.Round(time.Millisecond)` on all values, you will truncate everything to seconds + 3 decimal points and end up with this:

```
latency: 1m10.123s
latency: 26.550s
```

Reply:
If we round to milisecond we loose all the microsecond precision. I prefer to remove the `Truncate` and leave it as is. Let's see how it results.